slang.h

```diff
@@ -851,17 +849,23 @@ extern "C"
     struct ISlangWriter : public ISlangUnknown
     {
     public:
-            /** Write the formatted string with the used args. If returns SLANG_E_NOT_IMPLEMENTED, will use a default internal implementation, and call write with result
-            @param format Format string
-            @param args The arguments
-            @return SLANG_E_NOT_IMPLEMENTED if not implemented, SLANG_OK on success */
-        virtual SLANG_NO_THROW SlangResult SLANG_MCALL writeVaList(const char* format, va_list args) = 0;
+            /** Begin an append buffer.
+            NOTE! Only one append buffer can be active at any time.
+            @param maxNumChars The maximum of chars that will be appended
+            @returns The start of the buffer for appending to. */    
+        virtual SLANG_NO_THROW char* SLANG_MCALL beginAppendBuffer(size_t maxNumChars) = 0;
+            /** Ends the append buffer, and is equivalent to a write of the append buffer.
+            NOTE! That an endAppendBuffer is not necessary if there are no characters to write.
+            @param buffer is the start of the data to append and must be identical to last value returned from beginAppendBuffer
+            @param numChars must be a value less than or equal to what was returned from last call to beginAppendBuffer
+            @returns Result, will be SLANG_OK on success */
+        virtual SLANG_NO_THROW SlangResult SLANG_MCALL endAppendBuffer(char* buffer, size_t numChars) = 0;
```

Comment:
We don't need to drill down too far on the design of this API (since ultimately it is more for internal use than external), but I could imagine a further optimization would be to have `beginAppendBuffer` be able to return a smaller buffer than `maxNumChars` (`maxByteCount`? `maxSize`?) in cases where a particular writer has a fixed-size buffer that it uses internally.